docs/release.md

```diff
@@ -0,0 +1,34 @@
+# Release
+This document describes how to release the ClusterAPI BYOH infrastructure provider.
+
+
+### Current BYOH release: `v0.3.0`
+
+---
+##### Follow below steps to make a new release in BYOH
+1. Update `metadata.yaml` file if a new major/minor version is being released. This is not needed in case of patch release. Map the newly added version to supported ClusterAPI contract version.
+
+2. Create new tag from main branch. For example - `git tag v0.3.0`
```

Comment:
```suggestion
2. Create a new git tag from the main branch of the upstream git repo(vmware-tanzu/cluster-api-provider-bringyourownhost). For example - `git tag v0.3.0`
```

Reply:
Also, you can mention that we follow semantic versioning for releases and link to https://semver.org/